File: 136. Single Number/singleNumber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const singleNumber = (nums) => {
  let res = 0;

  for (let i = 0; i < nums.length; i++) {
    res = nums[i] ^ res;
  }

  return res;
};

const nums = [4, 1, 2, 1, 2];
console.log('singleNumber: ', singleNumber(nums));

File: 136. Single Number/singleNumber.py
class Solution:
  def singleNumber(self, nums: list[int]) -> int:
    res = 0

    for num in nums:
      res ^= num

    return res
  

nums = [4, 1, 2, 1, 2]
solution = Solution()
print('singleNumber: ', solution.singleNumber(nums))
